src/wallet/wallet.h

```diff
@@ -1544,5 +1542,162 @@ class AddSpendingKeyToWallet : public boost::static_visitor<KeyAddResult>
     KeyAddResult operator()(const libzcash::InvalidEncoding& no) const;
 };
 
+struct LogTx
+{
+    std::map<std::string, CAmount> balances;
+    enum TransactionStateType { TX_GENERATED, TX_MINED };
+    std::map<uint256, TransactionStateType> txstate;
+
+    std::string ValueFromAmount(const CAmount& amount)
```

Comment:
We should move the method from the RPC logic into a compilation unit that can be reached by both the server and wallet.